hibernate-core/src/main/java/org/hibernate/query/results/dynamic/DynamicFetchBuilderLegacy.java

```diff
@@ -44,39 +48,37 @@
  */
 public class DynamicFetchBuilderLegacy implements DynamicFetchBuilder, NativeQuery.FetchReturn, DynamicFetchBuilderContainer {
 
-	private static final String ELEMENT_PREFIX = CollectionPart.Nature.ELEMENT.getName() + ".";
-	private static final String INDEX_PREFIX = CollectionPart.Nature.INDEX.getName() + ".";
+	private static final String ELEMENT_PREFIX = "element.";
+	private static final int ELEMENT_PREFIX_LENGTH = 8;
 
 	private final String tableAlias;
 
 	private final String ownerTableAlias;
-	private final String fetchableName;
+	private final Fetchable fetchable;
 
 	private final List<String> columnNames;
-	private final Map<String, FetchBuilder> fetchBuilderMap;
+	private final Map<Fetchable, FetchBuilder> fetchBuilderMap;
 	private final DynamicResultBuilderEntityStandard resultBuilderEntity;
 
-	private LockMode lockMode;
```

Comment:
Please restore this and call `creationState.registerLockMode( tableAlias, lockMode );` in `buildFetch()` to properly pass through the configured lock mode.

Reply:
@beikov https://github.com/hibernate/hibernate-orm/pull/9217/commits/63df6f977d5d7745c1632cc458de2477aab6921a